api/src/main/java/net/md_5/bungee/api/connection/ProxiedPlayer.java

```diff
@@ -237,6 +237,37 @@
      */
     void sendTitle(Title title);
 
+    /**
+     * Sends a resource pack to this player.
+     *
+     * @param url the url of resource pack
+     * @param hash the resource pack hash
+     * @throws IllegalArgumentException if hash is longer than 40 characters
+     */
+    void sendResourcePack(String url, String hash) throws IllegalArgumentException;
+
+    /**
+     * Gets the player's currently loaded resource pack hash.
```

Comment:
What if they have no pack?

Reply:
On newer clients the response contains the hash.

Reply:
Yes, and what happens if they have no pack.

Does it return null?
Does it return ""?
Does it return "IHaveNoPack"?